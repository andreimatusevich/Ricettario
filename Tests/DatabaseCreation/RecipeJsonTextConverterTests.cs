using System;
using System.IO;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using ASP;
using Newtonsoft.Json;
using NUnit.Framework;
using Ricettario.Shared;
using Ricettario.Shared.Model;

namespace Tests.DatabaseCreation
{
    public class RecipeJsonTextConverterTests
    {
        [Test]
        [UseReporter(typeof(WinMergeReporter))]
        public void Simple()
        {
            var fileName = @"..\..\DatabaseCreation\Data\simple_recepie.txt";
            TestParse(fileName);
        }

        [Test]
        [UseReporter(typeof(WinMergeReporter))]
        public void GoogleDocs()
        {
            var fileName = @"..\..\DatabaseCreation\Data\google_docs_recepie.txt";
            TestParse(fileName);
        }

        [Test]
        [UseReporter(typeof(WinMergeReporter))]
        public void WithDescription()
        {
            var fileName = @"..\..\DatabaseCreation\Data\classic_buttermilk_waffles.txt";
            TestParse(fileName);
        }

        private static void TestParse(string fileName)
        {
            var lines = File.ReadAllLines(fileName);

            var parser = new RecipeTextConverter(new TestRecipeFactory());
            var recepie = parser.FromText(lines);

            var recepieJson = JsonConvert.SerializeObject(recepie, Formatting.Indented,
                new JsonSerializerSettings {NullValueHandling = NullValueHandling.Ignore});

            var recepieText = parser.ToText(recepie); 

            Approvals.Verify(recepieJson + Environment.NewLine + "-------------------" + Environment.NewLine + recepieText);
        }
    }


    public class TestRecipeFactory: IFactory<Recipe>
    {
        public Recipe Create()
        {
            return new Recipe()
            {
            };
        }
    }
}