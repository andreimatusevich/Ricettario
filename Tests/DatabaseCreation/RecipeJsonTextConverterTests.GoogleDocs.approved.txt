{
  "Version": "0001-01-01T00:00:00",
  "Name": "Roast Leg of Lamb",
  "Author": "Pegano",
  "RecipeIngredients": [
    "1 leg of lamb, 6-7 pounds, trimmed of all fat",
    "2 garlic doves, sliced",
    "3 tablespoons olive oil",
    "2 teaspoons rosemary",
    "Dash sea salt and pepper to taste",
    "3/4 cup no fat, low sodium chicken broth"
  ],
  "RecipeInstructions": [
    "Preheat oven to 400.°",
    "Make small slits Into the leg of lamb and Insert the garlic slices. Mix oil with rosemary, salt and pepper and rub all over the surface, Place in a roasting pan and place in oven, Reduce temperature to 375.°",
    "Roast for about 1½-2 hours. Instant meat thermometer inserted into thickest part of the meat should register 145° for medium and 155° for well done. Let stand 10 minutes before serving.",
    "Add broth to the pan stirring to melt the solidified juices, boil for a few; minutes to make a natural gravy."
  ],
  "RecipeYield": "Serves 6 to 8"
}
-------------------
Roast Leg of Lamb
Serves 6 to 8

Ingredients:
	1 leg of lamb, 6-7 pounds, trimmed of all fat
	2 garlic doves, sliced
	3 tablespoons olive oil
	2 teaspoons rosemary
	Dash sea salt and pepper to taste
	3/4 cup no fat, low sodium chicken broth

Directions:
	Preheat oven to 400.°
	Make small slits Into the leg of lamb and Insert the garlic slices. Mix oil with rosemary, salt and pepper and rub all over the surface, Place in a roasting pan and place in oven, Reduce temperature to 375.°
	Roast for about 1½-2 hours. Instant meat thermometer inserted into thickest part of the meat should register 145° for medium and 155° for well done. Let stand 10 minutes before serving.
	Add broth to the pan stirring to melt the solidified juices, boil for a few; minutes to make a natural gravy.

Description:

Author:
	Pegano

Tags: